test/test_torch.py

```diff
@@ -2519,6 +2526,57 @@ def test_serialization(self):
             self.assertEqual(c[1], c[3], 0)
             self.assertEqual(c[4], c[5][1:4], 0)
 
+            # check that serializing the same storage view object unpickles
+            # it as one object not two (and vice versa)
+            views = c[7]
+            self.assertEqual(views[0]._cdata, views[1]._cdata)
+            self.assertEqual(views[0], views[2])
+            self.assertNotEqual(views[0]._cdata, views[2]._cdata)
```

Comment:
I see. But we should add a test for deduplication of Python objects too. Can you add insert the same Python view into the list twice and check that it satisfies `assertIs` after loading?

Also, I see you've added a view with an offset of 0, but it's not used in tests.